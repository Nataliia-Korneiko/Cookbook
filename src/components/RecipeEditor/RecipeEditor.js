/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import LevelSelector from '../LevelSelector/LevelSelector';
import Level from '../../utils/Level';
import Button from '../ButtonEditor/ButtonEditor';
import s from './RecipeEditor.module.css';

const options = Object.values(Level);

export default class RecipeEditor extends Component {
  static propTypes = {
    onSave: PropTypes.func.isRequired,
    onCancel: PropTypes.func.isRequired,
    text: PropTypes.string,
    description: PropTypes.string,
    level: PropTypes.string,
    createDate: PropTypes.string,
    editDate: PropTypes.string,
  };

  static defaultProps = {
    text: '',
    description: '',
    level: Level.NORMAL,
    createDate: '',
    editDate: '',
  };

  state = {
    text: this.props.text,
    description: this.props.description,
    level: this.props.level,
    createDate: this.props.createDate,
    editDate: this.props.editDate,
  };

  handleChange = e => {
    this.setState({
      [e.target.name]: e.target.value,
    });
  };

  handleSubmit = e => {
    e.preventDefault();

    this.props.onSave({ ...this.state });

    this.setState({
      text: '',
      description: '',
      level: Level.NORMAL,
      createDate: '',
      editDate: '',
    });
  };

  render() {
    const { text, description, level } = this.state;
    const { onCancel } = this.props;

    return (
      <form className={s.form} onSubmit={this.handleSubmit}>
        <label className={s.label}>
          <input
            className={s.input}
            type="text"
            name="text"
            value={text}
            onChange={this.handleChange}
            placeholder="Enter the name of the recipe"
          />
        </label>

        <label className={`${s.label} ${s.label__description}`}>
          <textarea
            className={s.textarea}
            name="description"
            value={description}
            onChange={this.handleChange}
            placeholder="Enter the description for your recipe"
          />
        </label>

        <label className={`${s.label} ${s.label__level}`}>
          Select the recipe difficulty level:
          <LevelSelector
            options={options}
            value={level}
            onChange={this.handleChange}
          />
        </label>

        <div className={s.buttons}>
          <Button label="Save" type="submit" />
          <Button label="Cancel" type="button" onClick={onCancel} />
        </div>
      </form>
    );
  }
}
