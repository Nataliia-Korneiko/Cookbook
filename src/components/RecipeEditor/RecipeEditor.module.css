.form {
  padding: 30px 0;
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 620px;
}

.form::after {
  position: absolute;
  display: block;
  right: 0;
  bottom: 0;
  border-radius: 16px;
  content: '';
  width: 161.1px;
  height: 216px;
  background-image: url(../../assets/img/decoration/chef.png);
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
}

.label {
  margin-bottom: 20px;
  width: 500px;
}

.input {
  padding: 0 10px;
  display: block;
  width: 100%;
  height: 44px;
  border: 2px solid var(--color-golden);
  border-radius: 5px;
  font-size: 18px;
  color: var(--color-dark-blue);
  outline: none;
}

.textarea {
  padding: 0 60px 0 10px;
  display: block;
  width: 100%;
  height: 200px;
  border: 2px solid var(--color-golden);
  border-radius: 5px;
  font-size: 18px;
  color: var(--color-dark-blue);
  outline: none;
  resize: none;
}

.label__level {
  font-family: 'Dancing Script', cursive;
  font-size: 22px;
  font-weight: 700;
  color: var(--color-golden);
  align-items: center;
}

.label__level select {
  margin-left: 20px;
  height: 34px;
  width: 106px;
  border: transparent;
  font-family: 'Montserrat', sans-serif;
  font-size: 18px;
  background-color: var(--bg-color-golden);
  outline: none;
}

.buttons {
  padding-left: 60px;
  display: flex;
  justify-content: space-between;
  width: 440px;
  align-self: baseline;
}

@media (max-width: 980px) and (orientation: landscape) {
  .textarea {
    height: 80px;
  }
}

@media (max-width: 660px) and (orientation: portrait) {
  .form {
    width: 300px;
  }

  .form::after {
    bottom: 94px;
    width: 143px;
    height: 190px;
  }

  .label {
    width: 270px;
  }

  .label__description {
    margin-bottom: 30px;
  }

  .label__level {
    padding-right: 12px;
    font-size: 28px;
  }

  .label__level select {
    margin-top: 40px;
    margin-left: 0;
    width: 120px;
    height: 44px;
  }

  .textarea {
    padding-right: 10px;
    height: 160px;
  }

  .buttons {
    padding-left: 0;
    width: 270px;
    align-self: center;
  }
}

@media (max-width: 620px) and (orientation: landscape) {
  /* .form {
    overflow-x: scroll;
  } */

  /* .buttons {
    padding-left: calc((100vw - 500px) / 2);
    width: 414px;
  }

  .form::after {
    width: 138px;
    height: 183px;
  } */
}
