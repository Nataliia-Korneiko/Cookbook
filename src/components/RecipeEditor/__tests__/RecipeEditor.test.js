/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { shallow } from 'enzyme';
import RecipeEditor from '../RecipeEditor';
import LevelSelector from '../../LevelSelector/LevelSelector';

it('RecipeEditor renders a LevelSelector', () => {
  const wrapped = shallow(<RecipeEditor />);

  expect(wrapped.find(LevelSelector).length).toBe(1);
});

describe('RecipeEditor', () => {
  const props = {
    onSave: () => null,
    onCancel: () => null,
    text: '',
    description: '',
    level: '',
    createDate: '',
    editDate: '',
  };

  const newRecipeEditor = shallow(<RecipeEditor {...props} />);

  it('renders a properly', () => {
    expect(newRecipeEditor).toMatchSnapshot();
  });

  it('renders one form', () => {
    expect(newRecipeEditor.find('form.recipes__form')).toHaveLength(1);
  });

  it('renders three label', () => {
    expect(newRecipeEditor.find('label')).toHaveLength(3);
  });

  it('renders one input', () => {
    expect(newRecipeEditor.find('input[type="text"]')).toHaveLength(1);
  });

  it('renders one textarea', () => {
    expect(newRecipeEditor.find('textarea[name="description"]')).toHaveLength(
      1,
    );
  });

  it('renders one div', () => {
    expect(newRecipeEditor.find('div.recipes__buttons')).toHaveLength(1);
  });

  it('renders one button', () => {
    expect(newRecipeEditor.find('button[type="submit"]')).toHaveLength(1);
  });

  it('renders one button', () => {
    expect(newRecipeEditor.find('button[type="button"]')).toHaveLength(1);
  });
});
