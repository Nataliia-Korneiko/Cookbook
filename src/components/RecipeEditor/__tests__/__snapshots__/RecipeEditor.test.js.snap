// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeEditor renders a properly 1`] = `ShallowWrapper {}`;
